go/tools/builders/generate_checker_main.go

```diff
@@ -31,15 +31,206 @@ import (
 var codeTpl = `
 package main
 
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"go/ast"
+	"go/build"
+	"go/parser"
+	"go/token"
+	"go/types"
+	"os"
+	"path/filepath"
+	"strings"
+
+{{range $i, $importPath := .}}
+	_ "{{$importPath}}"
+{{- end}}
+
+	"github.com/bazelbuild/rules_go/analysis"
+	"golang.org/x/tools/go/gcexportdata"
+)
+
+type multiFlag []string
+
+func (m *multiFlag) String() string {
+	if len(*m) == 0 {
+		return ""
+	}
+	return fmt.Sprint(*m)
+}
+
+func (m *multiFlag) Set(v string) error {
+	(*m) = append(*m, v)
+	return nil
+}
+
+var (
+	archiveFiles multiFlag
```

Comment:
Style nit: we've been following this pattern for builders:

* `main` should just call `log.SetFlags(0)` and `log.SetPrefix("GoCheck: ")`, then a `run` function that takes `args []string` and returns `error`. `main` should call `log.Fatal(err)` if `run` fails.This makes it at least theoretically possible to test the `run` function (though we don't usually do that).
* Flags should be defined locally within the `run` function using `flag.NewFlagSet`. That's just helpful for reducing global state.